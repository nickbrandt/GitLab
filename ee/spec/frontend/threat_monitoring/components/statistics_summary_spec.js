import { shallowMount } from '@vue/test-utils';
import StatisticsSummary from 'ee/threat_monitoring/components/statistics_summary.vue';

describe('StatisticsSummary component', () => {
  let wrapper;

  const factory = (options) => {
    wrapper = shallowMount(StatisticsSummary, {
      stubs: { GlSingleStat: true },
      ...options,
    });
  };

  const findAnomalousStat = () => wrapper.findAll('glsinglestat-stub').at(0);
  const findNominalStat = () => wrapper.findAll('glsinglestat-stub').at(1);

  beforeEach(() => {
    factory({
      propsData: {
        data: {
          anomalous: { title: 'Anomalous', value: 0.2 },
          nominal: { title: 'Total', value: 100 },
        },
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders the anomalous traffic percentage', () => {
    expect(findAnomalousStat().element).toMatchSnapshot();
  });

  it('renders the nominal traffic count', () => {
    expect(findNominalStat().element).toMatchSnapshot();
  });
});
