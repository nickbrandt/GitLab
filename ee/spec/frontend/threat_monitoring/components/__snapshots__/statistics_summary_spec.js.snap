// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsSummary component renders the anomalous traffic percentage 1`] = `
<glsinglestat-stub
  class="col-sm-6 col-md-4 col-lg-3"
  title="Anomalous"
  unit="%"
  value="20"
  variant="warning"
/>
`;

exports[`StatisticsSummary component renders the nominal traffic count 1`] = `
<glsinglestat-stub
  class="col-sm-6 col-md-4 col-lg-3"
  title="Total"
  value="100"
/>
`;
